### PR TITLE
Va updates 2016 01 09

### PR DESCRIPTION
**Pull request by jteresco**
From @mapmikey by email.  Changes in active routes, some of which might need update entries.

Just looking for a quick look by a couple people before this goes live.

I-464 - centered point for 6A
US 1/301 - added new points at DupAve and RufMilRd north of Petersburg; moved the VA366 point (was at the wrong intersection) south of VA 10. (*newsworthy?)
US 11 - added a new point in Christiansburg (DepSt); added new point with VA 280 in Harrisonburg; moved US 11 off one-way pairs onto its current two-way routing in Winchester (*newsworthy?)
US 13 - the PitRd (first on north of NC line) waypoint is for a road that is on Google maps but does not actually connect to US 13.  I moved it to nearby Arthur Dr and renamed.  I should have left the old name as a hidden alternate and have attached the replacement US 13 file.
US 15 - changed SR643 to *SR643 since an overpass replaced the at-grade there (*newsworthy?)
US 17 - added a waypoint in Winchester where it is posted to end and also at Jubal Early Dr south of downtown (*newsworthy as change in terminus?)
US 33 - added a point at ParRd in Richmond
US 50 - changed Winchester routing (including adding a couple points) to its well-posted routing through downtown (*newsworthy)
US 58 - moved a point but did not rename in Norfolk where VA 337 splits; added a point for US 460 at Church St and renamed the previous US 460 point as just MonAv (don't know how to leave an alternate name if the alternate name is used again...) (*newsworthy - point in use moved)
US 460 - changed Petersburg routing to current use of I-95 to Wagner Rd; added point at SR 630 east of I-295; changed Norfolk routing to use Church instead of Monticello (*newsworthy)
US 460 ALT (Norfolk) - removed what was its eastern endpoint on TM which I don't think has ever been somewhere US 460 ALT went (always ended at Brambleton).  I did not extend it east along Brambleton to Church because while I think this might be its actual routing there is no proof in the field or on paper this was actually done; moved routing onto exclusively St. Paul's to connect with I-264. (*newsworthy)
US 460 Bus (Petersburg) - extended east out to US 460/VA 106 Wagner Rd (*newsworthy)
US 460 Bus (Christiansburg) - added a point at DepSt
US 522 - moved US 522 off one-way pairs onto its current two-way routing in Winchester plus new point at Jubal Early Dr south of downtown. (*newsworthy?)

### CODE REVIEW

**Comment by jteresco** [comment]
A few of these will need updates.csv entries.  I'm going to add "*newsworthy" to my original comment indicating which ones I think need an entry.

**Comment by yakra** [comment]
I've begun looking this over. I have some comments maybe 3/4 typed out on my PC at home; I should be able to finish it off in a couple hours.

**Comment by jteresco** [comment]
@yakra, thanks.

**Comment by yakra** [comment]
Nota bene: For point label change suggestions, I have not looked into whether AltLabels are required, unless AltLabels are specified.

**US1/301:**
The VA366 point is in use. It could be a newsworthy change; probably is, technically... There's no new label at the old location at Weir Rd. I don't know whether VA366 was ever located here in the past, but it could be that people who've used that point intended to indicate that they've used the actual location further south, and this was just the best approximation they had. As such it might even be able to squeak by without updates text, and I wouldn't shake my fist about it.
Of course, typing all that up probably took longer than just typing out some darn updates text. :)

**US11:**
US50/522_E -> US50/522
I'd have a go at writing updates text for the relocation in Winchester, but I can't figure out what the old alignment was. Was it Cameron northbound, and Braddock southbound? Fairmont-Picadilly-Braddock?

**US13:**
PitRd not in use; AltLabel unnecessary.
I'd put the ArtDr point [here](http://www.openstreetmap.org/?lat=36.565883&lon=-76.749909&zoom=15) instead.

**US15:**
Non-newsworthy.
'SR643Lee -> 'SR643_Lee (replace single quotes with asterisks. If I'd used actual asterisks here, GitHub Markdown would mangle my text.)

**US17:**
2016-01-15;(USA) Virginia;US 17;va.us017;Extended at north end from US 11 (south) to US 50 (west)

**US50:**
OK, lemme try this out:
2016-01-15;(USA) Virginia;US 50;va.us050;Removed from the Braddock St / Cameron St couplet, and relocated onto Braddock St, Cork St, and Cameron St between North Braddock St and US 11 south.
I'm only 99% sure this is correct, due to uncertainty over the old US11 routing (See above). Should be OK though.

**US58:**
US13/460_N -> US13/460
2016-01-15;(USA) Virginia;US 58;va.us058;Moved waypoint US460 from Monticello Ave to Church St

**US460:**
US58/460Alt_E -> US58/460Alt
VA337 -> VA337_W
VA337_N -> VA337_E
2016-01-15;(USA) Virginia;US 460;va.us460;Removed from US 460 Business (Petersburg) and relocated onto I-95 and Wagner Rd, between I-95 Exit 50 and VA 106. Removed from VA 337 and Monticello Ave and relocated onto Church St between VA337 east and Monticello Ave

**US460AltNor:**
US460_W -> US460 +US460_W
Before I try to tackle the updates text, would the old WB routing have been Monticello and City Hall?

**US460BusPet:**
I-95Con may be better off as I-95(50A). In which case I-95 -> I-95(52).
2016-01-15;(USA) Virginia;US 460 Business (Petersburg);va.us460buspet;Extended at east end from I-95 Exit 50A to Wagner Rd.
*(makes a bit more intuitive sense IMO if the above label change is implemented.)*

**US522:**
US17/50_S -> US17/50
I'd have a go at writing updates text for the relocation in Winchester, but I can't figure out what the old alignment was. Was it Cameron northbound, and Braddock southbound? Fairmont-Picadilly-Braddock?

-----

Directional suffixes, specifically could use a look over, other than just what I've listed above.

**Comment by mapmikey** [comment]
US 1/301:  VA 366 has always been in the location I moved the point to.  Weir Rd east of 1/301 used to only curl around to the north.  It wasn't until 2013 it could be used to reach the backend of VA 366.  I would surmise people used it not even noticing it was slightly north of the real VA 366 (and if this is ok, why would it matter if the very minor Weir Rd. would then be approximated by the nearby VA 366 point?).

US 11: I don't object to the label suggestion although the original one is still consistent with plenty of existing waypoints I've seen in active systems - it is still the EB US50 split.  In Winchester, US 11 NB is and was Valley/Gerrard/Cameron and US 11 SB used to be Loudoun/Wyck/Braddock.  For some reason 11 SB now uses Loudoun/Commercial/Cameron

US 13: I didn't check GMSV on that intersection.  I have moved the point to the actual intersection and re-submitted to Jim

US 15: All I did was put the asterisk.  The Lee without the underscore was previously the accepted name but if it must be changed I can do this.

US 58: waypoint suggestion: see my comment under US 11

US 460: I changed the 3rd waypoint which was one I added but the other 2 were previously accepted

US 460 ALT: you are correct on the old WB routing; the label is in accordance to convention for a bannered route, plus what happens if Norfolk signs it in a way it connects back to US 460?

US 460 Bus: I changed the I-95 labels (the connector part was once US 460 ALT a million years ago which is why I didn't think of it as a direct interchange).  Not sure why moving the waypoint is necessary...it splits between the way to get to/from Winfield Ave and was already accepted for whoever created the original data set.

US 522: The old SB route was Fairmont/Wyck/Braddock and the old NB route was Cameron/Wyck/Fairmont.  For the waypoint suggestion, see comment for US 11.

**Comment by jteresco** [comment]
I admit I have not paid attention to this.  What's our status on being able to get these changes in?

**Comment by yakra** [comment]
Forgot about this one. I'll have to give it a closer look...

**Comment by yakra** [comment]
Sorry I dropped the ball on this one for so long, guys. Here are my comments...
@mapmikey, does my proposed updates text check out?

--

**US1/301:** Sounds good. I'm OK with not calling this newsworthy, and relocating VA366 with no point at Weir Rd.

**US11:** The label suggestion is a style point coming from [the manual](http://cmap.m-plex.com/tools/manual_wayptlabels.php):
> Avoid extra distinguishing suffixes on labels with 2 numbered designations unless necessary. 
> Only in the case of two identical two-designation waypoints should the suffixes be added for distinction. Prefer the single-letter suffixes in this case.

There may be some in active systems (US hwys?) that aren't consistent with those rules -- it's likely they were created before those guidelines were codified.
2016-02-26;(USA) Virginia;US 11;va.us011;Southbound alignment removed from Loudon, Wyck and Braddock Streets and relocated onto Cameron Street between Commercial Street and Gerrard Street.

**US15:** Another semi-obscure style point. "SR643Lee" would indicate that there's a highway called "SR643Lee" *(so named in the HB)* that intersects here. "SR643_Lee" indicates that there's a highway called "SR643" intersecting here in "Lee" among other places.

**US460:** US58/460Alt_E -> US58/460Alt falls into the same category as the US11 & US58 relabeling suggestions. VA337, as a multiplex split, should receive a directional suffix.

**US460AltNor:** A letter-suffix convention would only apply when there's more than one intersection with US460. With just one connection here, plain US460 is preferred. Should Norfolk sign it in a way that it connects back to US 460, the two labels at this point would simply be swapped: US460_W would become the primary label again, and plain US460 would be the hidden alternate label instead.
2016-02-26;(USA) Virginia;US 460 Alternate (Norfolk);va.us460altnor;Truncated at east end from the Monticello Ave & St Pauls Blvd intersection to Brambleton Ave (VA 337). Westbound alignment removed from Monticello Ave and City Hall Ave and relocated onto St Pauls Blvd between Brambleton Ave (VA 337) and City Hall Ave.
*Just to make sure: The westbound alignment uses the City Hall Ave ramp, and not the Market St ramp paralleling the eastbound alignment?*

**US460BusPet:** No real biggie about repositioning the waypoint. I was thinking, put it where the road centerlines cross and disregard the turning lane. Feel free to write this off as me being overly anal.

**US 522:**
>The old SB route was Fairmont/Wyck/Braddock and the old NB route was Cameron/Wyck/Fairmont.

Ow, what a headache. K, uhh, here goes...
2016-02-26;(USA) Virginia;US 522;va.us522;Southbound alignment removed from Fairmont Ave, Wyck St and Braddock St and relocated onto Cameron St between Commercial Street and Gerrard Street. Northbound alignment removed from Wyck St and Fairmont Ave and relocated onto Cameron St between Wyck St and Commercial St.

The following points may be worthwhile:
**US522:** [Fairmont & Commercial](http://www.openstreetmap.org/?lat=39.195869&lon=-78.165316&zoom=15), as this is where both directions of the old alignment split with both directions of the new alignment.
**US11/522:** [Cameron & Wyck](http://www.openstreetmap.org/?lat=39.191882&lon=-78.161067&zoom=15), where the old NB alignment splits off the new alignment.
The reasoning here is lack of "pivot waypoints" bookending both ends of the relocation. (Admittedly, in cases of one-way couplets, this isn't always possible to do precisely.) The latter waypoint is arguably the less useful of the two, just being more useful for one direction of travel; there's room for differing opinions here. (See: my discussion with froggie on the TM forum re my proposed *OldI-93 point in NH, where I got really hung up on establishing my point of view as valid.) So if you really don't want to include this point (or, well, either of them, I guess), I can look the other way and not start a flame war, in deference to your maintenance of Virginia. :)

**Comment by mapmikey** [comment]
I did all the suggested waypoint changes except US460buspet and US460altnor for VA337 (this intersection does not start or end a multiplex - both routes have L shapes at this intersection).

I am attaching all the files I changed for this, plus adding the US460buslyn (added a waypoint for direct connection to US 460 freeway) and I-264 (new Exit 19C)...
[2-27-16 virginia update.zip](https://github.com/TravelMapping/HighwayData/files/149622/2-27-16.virginia.update.zip)

**Comment by jteresco** [comment]
The @mapmikey changes from the zip file are now in.  Let me know when this is good to merge in.  It shouldn't be hard to resolve whatever conflicts it's upset about if I do it at the command line.

**Comment by yakra** [comment]
Ah yes. The US460/VA337 bump looks OK to me then. Yes, let's go ahead with the ingestion of this update.

**Comment by jteresco** [comment]
Two conflicts I'm not sure how to resolve:

In VA 314, which of these is correct?

**Comment by mapmikey** [comment]
In both cases the second option is the correct one...

SprRd was renamed BeaWay very recently in honor of VTech Coach Beamer retiring and the VA 165-190 intersection moved after a rebuild of VA 165 through Kempsville

**Comment by yakra** [comment]
> @jteresco added a commit to jteresco/HighwayData that referenced this pull request 2 hours ago

Jim, how did you do that? When I try to reference a pull request or issue from a comment in yakra/HighwayData, the only option available AFAIK is to reference those in yakra/HighwayData rather than TravelMapping/HighwayData.

**Comment by jteresco** [comment]
I have a copy of jteresco/HighwayData that I mistakenly committed to first.  Then I re-did the merge and conflict resolution on TravelMapping/HighwayData.  Hopefully in the end everything worked out as intended.

**Comment by yakra** [comment]
I just meant, making a comment on GitHub that references an issue or pull.
How do you make a comment in one repo that references an issue/pull in another repo?
(Or do you yourself even know?)

**Comment by jteresco** [comment]
A very good question.  As with many things in version control, git in particular, it just happened.

**Comment by si404** [comment]
@yakra https://github.com/TravelMapping/HighwayData/pull/338#issuecomment-190256707 simply putting github.com/TravelMapping/UserData/pull/147 in the message, without removing the https:// like I did, gets you https://github.com/TravelMapping/UserData/pull/147 (the most recent pull request in that repository). Ditto github.com/si404/HighwayData/commit/2e62aac0dc5faa524b8afffd6a1c34b57d20047b becomes https://github.com/si404/HighwayData/commit/2e62aac0dc5faa524b8afffd6a1c34b57d20047b. And that first link, to your comment, was github.com/TravelMapping/HighwayData/pull/338#issuecomment-190256707 got by copying the link at the timestamp on your comment.

tl;dr - all github links are shortened.

**Comment by mapmikey** [comment]
Hi Jim--


  I have discovered a SC 34 TRUCK in Ridgeway.  See attached files:


new sc 34 TRUCK file

updated US 21 and SC 34 files

updated .csv files


Thanks-

Mike


________________________________

**Comment by jteresco** [comment]
@mapmikey looks like you meant to email me but instead emailed a github address that ended up commenting on this old pull request.  Please email me the new files at my regular address.
